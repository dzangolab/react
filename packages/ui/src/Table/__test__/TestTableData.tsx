const data = [
  { id: 1, name: "jone" },
  { id: 2, name: "abhi" },
  { id: 3, name: "mike" },
  { id: 4, name: "tony" },
];

const biggerData = [
  { id: 1, name: "jone" },
  { id: 2, name: "abhi" },
  { id: 3, name: "mike" },
  { id: 4, name: "tony" },
  { id: 5, name: "tony" },
  { id: 6, name: "tony" },
  { id: 7, name: "tony" },
  { id: 8, name: "tony" },
  { id: 9, name: "tony" },
  { id: 10, name: "tony" },
  { id: 11, name: "tony" },
  { id: 12, name: "tony" },
  { id: 13, name: "tony" },
  { id: 14, name: "tony" },
  { id: 15, name: "tony" },
];

const fetcher = () => {
  return data;
};

const columns = [
  {
    accessorKey: "id",
    header: () => <span>ID</span>,
  },
  {
    accessorKey: "name",
    header: () => <span>Name</span>,
  },
];

export { columns, data, fetcher, biggerData };
