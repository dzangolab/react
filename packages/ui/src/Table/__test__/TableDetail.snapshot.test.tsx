import React from "react";
import renderer, { create } from "react-test-renderer";
import { expect, test } from "vitest";

import { columns, data, fetcher } from "./TestTableData";
import { Table, TableDetail } from "../index";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  const component = create(
    <Table
      columns={columns}
      data={data}
      fetcher={fetcher}
      totalItems={data.length}
    >
      <TableDetail detail="Go to page:" showPrefix="Show" />
    </Table>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
