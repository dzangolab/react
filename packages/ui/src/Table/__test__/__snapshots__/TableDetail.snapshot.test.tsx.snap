// Vitest Snapshot v1

exports[`Component matches snapshot 1`] = `
<div>
  <span>
    <strong>
      1 of 1 
    </strong>
  </span>
  <div>
    <span>
      Go to page:
    </span>
    <input
      max={1}
      min={1}
      onBlur={[Function]}
      onChange={[Function]}
      onKeyDown={[Function]}
      type="number"
      value={1}
    />
    <select
      onChange={[Function]}
      value={10}
    >
      <option
        value={10}
      >
        Show
         
        10
      </option>
      <option
        value={20}
      >
        Show
         
        20
      </option>
      <option
        value={30}
      >
        Show
         
        30
      </option>
    </select>
  </div>
</div>
`;
